README.md

```diff
@@ -4,18 +4,58 @@
 [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
 ![npm](https://img.shields.io/npm/v/electron-ngrx?style=flat-square)
 
-**ElectronNgrx** is a library for dispatching NGRX actions between different Electron windows on a multi-window Electron Apps.
+## Motivation
+Synchronizing multiple Electron windows to a single state or in some cases multiple states is a difficult task which results in complex management and repetitive code.
 
-The library supports the following API methods:
+Implementing such a communication solution for multiple windows that transfer a decent amount of data between them takes a lot of effort, especially if you have multiple states that are shared.
+Basically you will find yourself building a rest API.
 
-`dispatchToParent` - Dispatch NGRX action data through the window's parent.
+### The solution
+ElectronNgrx offers an easy to use solution. It seamlessly dispatches actions and selects data from states across multiple windows.
 
-`dispatchToId`  - Dispatch NGRX action data to a specific Electron window by providing the destination window's ID.
+##Usge:
+ElectronNgrx delivers an Angular service with the following methods:
 
-`dispatchToRoute`  - Dispatch NGRX action data to a specific Route providing the destination route's details.
+`dispatchToParent` - Dispatch NGRX action to the window parent.
 
+`dispatchToId` - Dispatch NGRX action to a specific Electron window that matches the given id.
 
+`dispatchToRoute` - Dispatch NGRX action to all windows on the specific route.
 
-## Build
+`selectFromId` - select data from the state of the window that matches the given id.  
 
-Run `ng build` to build the project. The build artifacts will be stored in the `dist/` directory.
+`selectFromParent` - select data from parent window state. 
+
+## example
+for example if you want to increase counter on parent window state:
+```typescript
+export class Component {
+ 
+  constructor(private electronNgrxService: ElectronNgrxService) {
+  }
+
+  increaseCounterOnParentWindow(increaseBy) {
+    this.electronNgrxService.dispatchToParent(incrementAction({paylaod: increaseBy}));
+  }
+}
+```
+ElectronNgrx will send the action to parent window and dispatch the action to the state and trigger the Angular change detection on the parent window. 
+
+## Demo
+
+To clone and run the demo you'll need [Git](https://git-scm.com) and [Node.js](https://nodejs.org/en/download/) (which comes with [npm](http://npmjs.com)) installed on your computer. From your command line:
+
+```bash
+# Clone this repository
+git clone https://github.com/meniRoy/electron-ngrx.git
+# Go into the repository
+cd electron-ngrx
+# Install dependencies
+npm install
+# Run the the demo
+npm start
+```
+## change detection
+By default ngZone dont know about ipc electron.
```

Comment:
By default NgZone isn't aware of the ipc electron.
This may cause issues when messages from electron ipc are received and the change detection isn't triggered.
ElectronNgrx takes care of this problem by triggering the Angular change detection after every action that affects the window such as dispatching an action received from another window or receiving data from another window's state.